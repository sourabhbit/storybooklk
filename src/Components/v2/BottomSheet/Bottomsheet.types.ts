export interface BottomsheetTypes {
	children?: React.ReactNode
	expanded: boolean
	backdrop?: boolean
	scrollable: boolean
	onClickBackdrop?: () => void
}
