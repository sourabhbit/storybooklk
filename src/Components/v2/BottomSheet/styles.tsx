import styled from '@emotion/styled-base'

export const BottomSheetContainer = styled('div')`
	position: fixed;
	left: 0;
	top: 0;
	width: 100%;
	height: 100%;
	overflow: scoll;
	pointer-events: none;
	z-index: 200;
`

export const BackdropWrapper = styled('div')<{
	expanded: boolean | undefined
}>`
	transition: opacity 0.45s;
	position: fixed;
	top: 0;
	bottom: 0;
	left: 0;
	right: 0;
	z-index: 50;
	background-color: ${(props) =>
		props.expanded ? 'rgba(33, 33, 33, 0.75)' : ''};
	pointer-events: ${(props) => (props.expanded ? 'auto' : 'none')};
`

export const BottomSheet = styled('div')<{
	expanded: boolean
	scrollable: boolean
}>`
	position: absolute;
	overflow: ${(props) => (props.scrollable ? 'scroll' : 'hidden')};
	left: 0;
	right: 0;
	bottom: 0;
	pointer-events: none;
	background: white;
	-webkit-transform: translate3d(0, 100%, 0);
	-webkit-transition-duration: 0.4s;
	-webkit-transition-property: -webkit-transform;
	-webkit-transition-timing-function: cubic-bezier(0.25, 0.8, 0.25, 1);
	transition-duration: 0.4s;
	transition-property: transform;
	transition-timing-function: cubic-bezier(0.25, 0.8, 0.25, 1);
	-webkit-transform: translate3d(0, 0, 0);
	transform: ${(props) =>
		props.expanded ? 'translate3d(0, 0, 0)' : 'translate3d(0, 100%, 0)'};
	pointer-events: ${(props) => (props.expanded ? 'auto' : 'none')};
	z-index: ${(props) => (props.expanded ? '101' : '201')};
`
