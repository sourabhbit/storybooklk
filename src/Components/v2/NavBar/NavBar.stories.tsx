import { ComponentMeta, ComponentStory } from '@storybook/react'
import { generateArgTypes } from '../../../Helpers/Storybook'
import NavBar from './index'

export default {
	title: 'Components/v2/NavBar',
	component: NavBar,
	argTypes: {
		options: generateArgTypes({
			type: 'array',
			required: true,
			description: 'Show dropdown options',
			category: 'Component',
		}),
	},
} as ComponentMeta<typeof NavBar>

const Template: ComponentStory<typeof NavBar> = () => (
	<NavBar
		options={[
			'Your Picks',
			'Discover',
			'Categories',
			'Gold',
			'Collabs',
			'Labs',
		]}
	/>
)

export const Overview = Template.bind({})
