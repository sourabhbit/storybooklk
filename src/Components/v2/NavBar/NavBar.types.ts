export interface NavBarTypes {
	options?: string[]
}
