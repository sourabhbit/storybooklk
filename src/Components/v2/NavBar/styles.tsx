import styled from 'styled-components'

export const NavBarWrapper = styled.div`
	display: flex;
	cursor: pointer;
`
