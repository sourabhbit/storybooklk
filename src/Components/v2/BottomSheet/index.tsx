import { BottomSheetContainer, BackdropWrapper, BottomSheet } from './styles'
import { BottomsheetTypes } from './Bottomsheet.types'

const Bottomsheet = ({
	backdrop,
	expanded,
	onClickBackdrop = () => null,
	children,
	scrollable,
}: BottomsheetTypes) => {
	return (
		<BottomSheetContainer
			data-testid="bottomsheet-container"
			aria-label="bottom-sheet-container"
		>
			<>
				{backdrop && expanded && (
					<BackdropWrapper
						data-testid="backdrop-container"
						expanded={expanded}
						aria-label="backdrop"
						onClick={() => onClickBackdrop()}
					></BackdropWrapper>
				)}
				<BottomSheet
					expanded={expanded}
					scrollable={scrollable}
					aria-label="bottom-sheet"
				>
					{children}
				</BottomSheet>
			</>
		</BottomSheetContainer>
	)
}

export default Bottomsheet
