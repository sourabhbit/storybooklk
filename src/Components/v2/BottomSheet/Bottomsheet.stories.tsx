import { ComponentMeta, ComponentStory } from '@storybook/react'
import NavBar from 'Components/v2/NavBar'
import { generateArgTypes } from '../../../Helpers/Storybook'
import Bottomsheet from './index'

export default {
	title: 'Components/v2/Bottomsheet',
	component: Bottomsheet,
	argTypes: {
		children: {
			table: {
				category: 'Children',
			},
		},
		expanded: generateArgTypes({
			type: 'boolean',
			required: false,
			description: 'expand should available',
			category: 'Component',
		}),
		backdrop: generateArgTypes({
			type: 'boolean',
			required: false,
			description: 'backdrop should available',
			category: 'Component',
		}),
		scrollable: generateArgTypes({
			type: 'boolean',
			required: false,
			description: 'if we wish to add scroll in bottomsheet',
			category: 'Component',
		}),
		onClickBackdrop: generateArgTypes({
			type: 'function',
			required: false,
			description: 'onClick handler for filter',
			category: 'Events',
		}),
	},
} as ComponentMeta<typeof Bottomsheet>

const Template: ComponentStory<typeof Bottomsheet> = (args) => (
	<Bottomsheet {...args} />
)

export const Overview = Template.bind({})
Overview.args = {
	children: (
		<p style={{ background: 'white' }}>
			<NavBar
				options={[
					'Your Picks',
					'Discover',
					'Categories',
					'Gold',
					'Collabs',
					'Labs',
				]}
			/>
		</p>
	),
	expanded: true,
	backdrop: true,
	scrollable: true,
	onClickBackdrop: () => {
		expanded: false
	},
}
