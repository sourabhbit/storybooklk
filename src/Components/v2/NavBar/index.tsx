import { NavBarWrapper } from './styles'
import { NavBarTypes } from './NavBar.types'
import { ComponentSizeENUM, TypographyENUM } from '../../../Types/general'
import NavButton from '../NavButton'
import React from 'react'

const NavBar = ({ options }: NavBarTypes) => {
	const [selectedNav, setSelectedNav] = React.useState(0)

	const changeNav = (index: number) => {
		setSelectedNav(index)
	}
	return (
		<NavBarWrapper data-testid="navbar-wrapper">
			{options?.map((nav, index) => (
				<NavButton
					selectedNav={selectedNav}
					onClick={() => changeNav(index)}
					index={index}
					text={nav}
					key={index}
					id="id"
					componentSize={ComponentSizeENUM.small}
					font={TypographyENUM.serif}
				></NavButton>
			))}
		</NavBarWrapper>
	)
}

export default NavBar
