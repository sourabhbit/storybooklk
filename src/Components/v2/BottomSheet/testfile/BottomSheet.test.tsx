import '@testing-library/jest-dom'
import { render, screen /*, fireEvent*/ } from '@testing-library/react'
import NavBar from 'Components/v2/NavBar'
import BottomSheet from '../index'

const props = {
	children: (
		<p style={{ background: 'white' }}>
			<NavBar
				options={[
					'Your Picks',
					'Discover',
					'Categories',
					'Gold',
					'Collabs',
					'Labs',
				]}
			/>
		</p>
	),
	expanded: true,
	backdrop: true,
	scrollable: true,
	onClickBackdrop: () => null,
}

describe('<BottomSheet />', () => {
	let _asFragment: any = null
	let _rerender: any = null

	beforeEach(() => {
		const { asFragment, rerender } = render(<BottomSheet {...props} />)
		_asFragment = asFragment
		_rerender = rerender
	})

	afterAll(() => {
		_asFragment = null
	})

	it('Should render component', () => {
		const bottomsheet = screen.queryByTestId('bottomsheet-container')
		const backdrop = screen.queryByTestId('backdrop-container')
		expect(bottomsheet).toBeInTheDocument()
		expect(backdrop).toBeInTheDocument()
	})

	// it('should call function ', () => {
	//     fireEvent.click(screen.getByTestId('backdrop-container'))
	// 	expect(props.onClickBackdrop).toHaveBeenCalledTimes(1)
	// })

	it('should not render component', () => {
		_rerender(<BottomSheet {...props} expanded={false} />)
		const backdrop = screen.queryByTestId('backdrop-container')
		expect(backdrop).not.toBeInTheDocument()
	})

	it('Should take a snapshot', () => {
		expect(_asFragment()).toMatchSnapshot()
	})
})
