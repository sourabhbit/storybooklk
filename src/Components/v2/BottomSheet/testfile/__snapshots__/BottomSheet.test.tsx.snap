// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomSheet /> Should take a snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="bottom-sheet-container"
    class="css-zlu9tc"
    data-testid="bottomsheet-container"
  >
    <div
      aria-label="backdrop"
      class="css-1fifkz0"
      data-testid="backdrop-container"
    />
    <div
      aria-label="bottom-sheet"
      class="css-1tqv9wj"
    >
      <p
        style="background: white;"
      />
      <div
        class="sc-bdvvtL efRztj"
        data-testid="navbar-wrapper"
      >
        <div
          class="sc-gsDKAQ cmHcgF"
          data-testid="nav-button"
          id="id"
        >
          <span
            class="sc-dkPtRN eNtPVD"
            data-testid="nav-text"
          >
            Your Picks
          </span>
        </div>
        <div
          class="sc-gsDKAQ kjYOaa"
          data-testid="nav-button"
          id="id"
        >
          <span
            class="sc-dkPtRN fbKlCn"
            data-testid="nav-text"
          >
            Discover
          </span>
        </div>
        <div
          class="sc-gsDKAQ kjYOaa"
          data-testid="nav-button"
          id="id"
        >
          <span
            class="sc-dkPtRN fbKlCn"
            data-testid="nav-text"
          >
            Categories
          </span>
        </div>
        <div
          class="sc-gsDKAQ kjYOaa"
          data-testid="nav-button"
          id="id"
        >
          <span
            class="sc-dkPtRN fbKlCn"
            data-testid="nav-text"
          >
            Gold
          </span>
        </div>
        <div
          class="sc-gsDKAQ kjYOaa"
          data-testid="nav-button"
          id="id"
        >
          <span
            class="sc-dkPtRN fbKlCn"
            data-testid="nav-text"
          >
            Collabs
          </span>
        </div>
        <div
          class="sc-gsDKAQ kjYOaa"
          data-testid="nav-button"
          id="id"
        >
          <span
            class="sc-dkPtRN fbKlCn"
            data-testid="nav-text"
          >
            Labs
          </span>
        </div>
      </div>
      <p />
    </div>
  </div>
</DocumentFragment>
`;
